grammarinator/runtime/cooldown_model.py

```diff
@@ -15,11 +15,18 @@ def __init__(self, model, weights=None, cooldown=1.0):
 
     def choice(self, node, idx, choices):
         i = self._model.choice(node, idx, [w * self._weights.get((node.name, idx, i), 1) for i, w in enumerate(choices)])
-        self._weights[(node.name, idx, i)] = self._weights.get((node.name, idx, i), 1) * self._cooldown
+        weights = CooldownModel.normalize([self._weights.get((node.name, idx, i), 1) * self._cooldown if iw == i else w for iw, w in enumerate(choices)])
```

Comment:
Is this right? The list of values that gets normalized is a mixed one. One of the values comes from `self._weights` but the rest comes from `choices`. But these are different values. The values in `choices` come from the predicates in the grammar. The values in `self._weights` start of with 1 and get "cooled down" -- and they are used as scaling factors to the values in `choices`. I guess that is should be only these latter values (i.e., those in `self._weights`) that should be normalized.

Moreover, I'm not sure that normalization deserves its own function as it is not used anywhere else.

So, shouldn't `choice` be rather something like:
```py
def choice(self, node, idx, choices):
    c = self._model.choice(node, idx, [w * self._weights.get((node.name, idx, i), 1) for i, w in enumerate(choices)])

    self._weights[(node.name, idx, c)] = self._weights.get((node.name, idx, c), 1) * self._cooldown
    wsum = sum(self._weights.get((node.name, idx, i)) for i in range(len(choices)))
    for i in range(len(choices)):
        self._weights[(node.name, idx, i)] = self._weights.get((node.name, idx, i), 1) / wsum

    return c
```
?

Reply:
Indeed, the non-cooldowned values are wrong. I'll fix it. On the other hand, normalization was put into a separate function, since it will be used at multiple places in the near future. But I can inline if for now, ofc.